yolact.py

```diff
@@ -462,40 +464,66 @@ def __init__(self):
             # This comes from the smallest layer selected
             # Also note that cfg.num_classes includes background
             self.class_existence_fc = nn.Linear(src_channels[-1], cfg.num_classes - 1)
-        
+
         if cfg.use_semantic_segmentation_loss:
             self.semantic_seg_conv = nn.Conv2d(src_channels[0], cfg.num_classes-1, kernel_size=1)
 
         # For use in evaluation
         self.detect = Detect(cfg.num_classes, bkg_label=0, top_k=cfg.nms_top_k,
             conf_thresh=cfg.nms_conf_thresh, nms_thresh=cfg.nms_thresh)
 
-    def save_weights(self, path):
+    def save_checkpoint(self, path, recipe=None, epoch=0):
         """ Saves the model's weights using compression because the file sizes were getting too big. """
-        torch.save(self.state_dict(), path)
-    
-    def load_weights(self, path):
-        """ Loads weights from a compressed save file. """
-        state_dict = torch.load(path)
+        checkpoint = {
+                         'epoch': epoch,
+                         'model_state_dict': self.state_dict(),
+                         'recipe': recipe,
+                     }
+        torch.save(checkpoint, path)
+
+    def load_checkpoint(self, path, recipe=None) -> Tuple[Optional[int], Optional[str], SparseMLWrapper]:
+        """ Loads weights into the current Yolact object.
+
+        :return: A tuple containing the epoch and the recipe if
+            found in the checkpoint, and a SparseML Wrapper instance
+        """
+
+        checkpoint = torch.load(path)
+        epoch = checkpoint.get('epoch')
+        recipe = recipe or checkpoint.get('recipe')
+        state_dict = checkpoint.get('model_state_dict', checkpoint)
+        sparseml_wrapper = SparseMLWrapper(self, recipe)
 
         # For backward compatability, remove these (the new variable is called layers)
         for key in list(state_dict.keys()):
             if key.startswith('backbone.layer') and not key.startswith('backbone.layers'):
                 del state_dict[key]
-        
+
             # Also for backward compatibility with v1.0 weights, do this check
             if key.startswith('fpn.downsample_layers.'):
                 if cfg.fpn is not None and int(key.split('.')[2]) >= cfg.fpn.num_downsample:
                     del state_dict[key]
-        self.load_state_dict(state_dict)
+
+        loaded = False
```

Comment:
commenting that we discussed changes to this section offline

Reply:
Changes Made!